common/lib/xmodule/xmodule/modulestore/tests/factories.py

```diff
@@ -22,6 +22,33 @@ class Dummy(object):
     pass
 
 
+class XModuleFactoryLock(threading.local):
```

Comment:
Are tests ever run on separate threads? Does it happen in Jenkins? What problem does this class solve?